### PR TITLE
Update Shadow CMake configuration for FreeRTOS console

### PR DESCRIPTION
**Pull request by aggarw13**
Changes include:
* Update Device Shadow cmake configuration for FreeRTOS console requirements by following `core_mqtt_demo_dependencies.cmake` file
* Add dependency of **coreMQTT demo** on `AFR::core_mqtt_demo_dependencies` module so that FreeRTOS console ONLY displays the core MQTT demo when the FreeRTOS console displayed "core MQTT" library option is selected. (`AFR::core_mqtt_demo_dependencies` represents the FreeRTOS console-aware CMake target for coreMQTT library, and similarly, `AFR::device_shadow_demo_dependencies` for the Device Shadow library`)
* Similar change of adding `AFR::device_shadow_demo_dependencies` dependency on the Device Shadow demo.

### CODE REVIEW

**Review by abhidixi11 — DISMISSED** [review]

File: libraries/device_shadow_demo_dependencies.cmake [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    PUBLIC
+        AFR::logging
+)
+
+################################################################
+
+# device_shadow_demo_dependencies module.
+# Metadata module used for the Device Shadow library in the FreeRTOS console.
+# It represents a collection of module dependencies required 
+# by the coreMQTT demos.
```
Comment (leegeth) [inline libraries/device_shadow_demo_dependencies.cmake:46]: shadow demos?

File: libraries/device_shadow_demo_dependencies.cmake [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+)
+
+################################################################
+
+# device_shadow_demo_dependencies module.
+# Metadata module used for the Device Shadow library in the FreeRTOS console.
+# It represents a collection of module dependencies required 
+# by the coreMQTT demos. 
+# This module enables the FreeRTOS console experience of enabling
+# library dependencies when selecting the MQTT library, so that the
```
Comment (leegeth) [inline libraries/device_shadow_demo_dependencies.cmake:48]: Shadow?

File: libraries/device_shadow_demo_dependencies.cmake [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+# coreMQTT demos can be downloaded.
```
Comment (leegeth) [inline libraries/device_shadow_demo_dependencies.cmake:49]: Same comment here as well.

File: libraries/device_shadow_demo_dependencies.cmake [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+afr_set_lib_metadata(DISPLAY_NAME "Device Shadow")
+afr_set_lib_metadata(CATEGORY "Connectivity")
+afr_set_lib_metadata(VERSION "1.0.0")
+afr_set_lib_metadata(IS_VISIBLE "true")
+
+afr_module_sources(
+    ${AFR_CURRENT_MODULE}
+    PRIVATE
+        # Adding sources so that CMake can generate the 
+        # core_mqtt_demo_dependencies target; otherwise, it gives the
```
Comment (leegeth) [inline libraries/device_shadow_demo_dependencies.cmake:64]: device_shadow_demo_dependencies?

File: libraries/device_shadow_demo_dependencies.cmake (hunk 1 above)
Comment (aggarw13) [reply libraries/device_shadow_demo_dependencies.cmake]: Fixed

File: libraries/device_shadow_demo_dependencies.cmake (hunk 2 above)
Comment (aggarw13) [reply libraries/device_shadow_demo_dependencies.cmake]: Fixed

**Review by leegeth — DISMISSED** [review]

File: libraries/device_shadow_demo_dependencies.cmake [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+# It represents a collection of module dependencies required 
+# by the Device Shadow demo. 
+# This module enables the FreeRTOS console experience of enabling
+# library dependencies when selecting the Device Shadow library, so that the
+# Device Shadow demo can be downloaded.
+afr_module(NAME device_shadow_demo_dependencies )
+
+afr_set_lib_metadata(ID "device_shadow_demo_dependencies")
+afr_set_lib_metadata(DESCRIPTION "The AWS IoT Device Shadow library enables you to \
+store and retrieve the current state (the “shadow”) of every registered device.")
```
Comment (aggarg) [inline libraries/device_shadow_demo_dependencies.cmake:54]: The double quote character seems weird. Please delete and re-type manually.

**Review by aggarg — APPROVED** [review]

**Review by leegeth — APPROVED** [review]

**Pull request merged**